### PR TITLE
SchemaOptions.safe can also be an object

### PR DESCRIPTION
**Pull request by dhritzkiv**
See: [safe docs](http://mongoosejs.com/docs/guide.html#safe)

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/mongoose/index.d.ts
Comment (sindrenm) [inline types/mongoose/index.d.ts]: I think there are more options that can be added here as well, however I wasn't able to find it after a (very quick) search on Google and the MongoDB docs.

Perhaps we should extract that into an interface instead and see if we can find a full list of options?

The link provided here is dead, it seems: http://mongoosejs.com/docs/2.7.x/docs/schema-options.html

**Review by sindrenm — COMMENTED** [review]
It would be great if you could add a test as well! :smile:

File: types/mongoose/index.d.ts
Comment (dhritzkiv) [reply types/mongoose/index.d.ts]: I believe this is the documentation/options available for the `safe` option: https://docs.mongodb.com/manual/reference/write-concern/index.html

**Comment by dhritzkiv** [comment]
Re. tests: I'm not sure what to add; the tests already seem to have a `safe` option with `option`:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/acc37d8b9d48ea194fcf9314cff2ab3cd5c5fe3a/types/mongoose/mongoose-tests.ts#L1258-L1259

Or am I missing something?

**Comment by sindrenm** [comment]
Oh my! You are absolutely right. Shame on me for not looking for that; I just assumed that to be part of this PR. That's my bad.

**Review by sindrenm — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Pull request merged**
